zeppelin-server/src/main/java/org/apache/zeppelin/rest/InterpreterRestApi.java

```diff
@@ -115,9 +116,10 @@ public Response updateSetting(String message, @PathParam("settingId") String set
           UpdateInterpreterSettingRequest.class);
       interpreterFactory.setPropertyAndRestart(settingId, p.getOption(), p.getProperties());
     } catch (InterpreterException e) {
-      return new JsonResponse(Status.NOT_FOUND, e.getMessage(), e).build();
+      return new JsonResponse(Status.NOT_FOUND, e.getMessage(), settingId).build();
```

Comment:
We can, but I don't think it would be useful here. In this code path, `InterpreterException` is only thrown when the interpreter setting id is not found.